### PR TITLE
chore: Make Feast UI importable as a NPM Module

### PR DESCRIPTION
**Pull request by tonyhschu**
**What this PR does / why we need it**:
Makes the Feast UI importable as a NPM Module. This will make it easier for customers that want to extend Feast UI to have maintain their own repo while still consuming our changes.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: ui/.gitignore [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (adchia) [inline ui/.gitignore:1]: I think this already exists in the parent directory? maybe revert or merge if there is more stuff?

File: ui/package.json [hunk 2]
```diff
@@ -1,37 +1,58 @@
...
+  "main": "./dist/feast-ui.cjs",
+  "module": "./dist/feast-ui.module.js",
+  "peerDependencies": {
     "@elastic/datemath": "^5.0.3",
-    "@elastic/eui": "^50.0.0",
-    "@emotion/react": "^11.8.1",
-    "@testing-library/jest-dom": "^5.16.2",
-    "@testing-library/react": "^12.1.3",
-    "@testing-library/user-event": "^13.5.0",
+    "@elastic/eui": "^46.1.0",
```
Comment (adchia) [inline ui/package.json:12]: any reason why we're using some old versions?

File: ui/src/components/FeaturesListDisplay.tsx [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 import SparklineHistogram from "./SparklineHistogram";
+import FeatureFlagsContext from "../contexts/FeatureFlagsContext";
 
 interface FeaturesListProps {
   featureViewName: string;
   features: FeastFeatureColumnType[];
 }
 
 const FeaturesList = ({ featureViewName, features }: FeaturesListProps) => {
+  const { enabledFeatureStatistics } = useContext(FeatureFlagsContext);
```
Comment (adchia) [inline ui/src/components/FeaturesListDisplay.tsx:14]: is this intentional? seems not used so maybe can revert

File: ui/src/contexts/FeatureFlagsContext.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+
+interface FeatureFlags {
```
Comment (adchia) [inline ui/src/contexts/FeatureFlagsContext.ts:3]: does this let users flip it on / off in the top level FeastUI component?

Comment (adchia) [inline ui/src/contexts/FeatureFlagsContext.ts:3]: does this let users flip it on / off in the top level FeastUI component?

File: ui/src/pages/Layout.tsx [hunk 5]
```diff
@@ -8,15 +8,20 @@ import {
...
-import feastWorkMark from "../feast-workmark.svg";
+import { useParams } from "react-router-dom";
+import { useLoadProjectsList } from "../contexts/ProjectListContext";
+
+import ProjectSelector from "../components/ProjectSelector";
+import Sidebar from "./Sidebar";
+import FeastWordMark from "../graphics/FeastWordMark";
 
 const Layout = () => {
+  // Registery Path Context has to be inside Layout
```
Comment (adchia) [inline ui/src/pages/Layout.tsx:22]: typo

Comment (adchia) [inline ui/src/pages/Layout.tsx:22]: typo

File: ui/src/FeastUISansProviders.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+} from "./contexts/ProjectListContext";
+
+interface FeastUIConfigs {
+  tabsRegistry: FeastTabsRegistryInterface;
+  featureFlags?: FeatureFlags;
+  projectListPromise?: Promise<any>;
+}
+
+const defaultProjectListPromise = () => {
+  return fetch("/projects-list.json", {
```
Comment (adchia) [inline ui/src/FeastUISansProviders.tsx:43]: seems more ideal to actually expose this as a parameter they can provide? 

e.g. a user may want this list to not exist locally and instead remotely on e.g. S3

Comment (adchia) [inline ui/src/FeastUISansProviders.tsx:43]: seems more ideal to actually expose this as a parameter they can provide? 

e.g. a user may want this list to not exist locally and instead remotely on e.g. S3

File: ui/package.json (hunk 2 above)
Comment (tonyhschu) [reply ui/package.json:12]: This is simply the version we started with during the hackathon. I'm not sure what is the right thing to do however, b/c:

1. On the one hand, EUI is a peer dependency, which means that customers could potentially use different versions, and we want our package to play nice with others.
2. On the other hand ... new updates to EUI might break our module. 

I'm not really sure what the best practice here is.

File: ui/src/components/FeaturesListDisplay.tsx (hunk 3 above)
Comment (tonyhschu) [reply ui/src/components/FeaturesListDisplay.tsx:14]: This is intentional. This enables the user to turn on this on as a prop on `<FeastUI>` i.e. 

```jsx
<FeastUI
  feastUIConfigs={{
    featureFlags: {
      enabledFeatureStatistics: true
    },
  }}
/>
```

This already works. I need to document this better hahaha.

File: ui/src/contexts/FeatureFlagsContext.ts (hunk 4 above)
Comment (tonyhschu) [reply ui/src/contexts/FeatureFlagsContext.ts:3]: Yup

Comment (tonyhschu) [reply ui/src/contexts/FeatureFlagsContext.ts:3]: See this comment: https://github.com/feast-dev/feast/pull/2370#discussion_r825364792

File: ui/src/pages/Layout.tsx (hunk 5 above)
Comment (tonyhschu) [reply ui/src/pages/Layout.tsx:22]: Fixed.

File: ui/src/FeastUISansProviders.tsx (hunk 6 above)
Comment (tonyhschu) [reply ui/src/FeastUISansProviders.tsx:43]: Yup, again another thing I need to document. You can override it like this.
```jsx
<FeastUI
  feastUIConfigs={{
    projectListPromise: fetch(SOME_PATH, {
      headers: {
        "Content-Type": "application/json",
      },
    }).then((res) => {
      return res.json();
    })
  }}
/>
```

**Comment by woop** [comment]
Is this really a "feature"? Seems more like a chore or housekeeping

**Comment by tonyhschu** [comment]
I didn't know what to pick. Happy to go with whatever @adchia thinks is more appropriate.

File: ui/.gitignore (hunk 1 above)
Comment (tonyhschu) [reply ui/.gitignore]: Yup. done.

**Comment by adchia** [comment]
Couple things to fix:
- Make the version 0.19.4 (our next patch release) + publish this
- In a followup PR, add documentation about publishing this package (add me to the feast org too)

**Comment by adchia** [comment]
Also sign the commits

**Comment by adchia** [comment]
Also: revert the deleting of the App.test.tsx

**Comment by woop** [comment]
> Couple things to fix:
> 
> * Make the version 0.19.4 (our next patch release) + publish this
> * In a followup PR, add documentation about publishing this package (add me to the feast org too)

Any reason we can't do the following instead

1. Make the version the same as other versions on master ([0.19.0](https://github.com/feast-dev/feast/blob/b35e1e84720523cef70cba6d6306af8f193b469f/java/pom.xml#L41)) and merge this PR in
2. Add packages.json to https://github.com/feast-dev/feast/blob/master/infra/scripts/release/files_to_bump.txt
3. Backport both (1) and (2) to v0.19-branch
4. Change the version in packages.json on the v0.19-branch to 0.19.3
5. Cut release on v0.19-branch, which will automatically bump packages.json to 0.19.4
(eventually) 6. Cut release on master, which will automatically bump packages.json to 0.20.0

The versioning script will then update from 0.19.0 to 0.1

**Comment by woop** [comment]
@adchia @tonyhschu next steps here?

**Comment by tonyhschu** [comment]
Hey @woop this is on me. I can follow the scheme you laid out above. I owe @adchia a couple other small changes - eg documenting how to publish to NPM.

**Review by adchia — APPROVED** [review]
/lgtm

**Pull request merged**